crates/ruff_python_parser/src/lexer/indentation.rs

```diff
@@ -124,6 +124,14 @@ impl Indentations {
         static ROOT: Indentation = Indentation::root();
         self.stack.last().unwrap_or(&ROOT)
     }
+
+    pub(super) fn len(&self) -> usize {
+        self.stack.len()
+    }
+
+    pub(super) fn truncate(&mut self, len: usize) {
+        self.stack.truncate(len)
+    }
```

Comment:
Nit: We may want to use newtype wrappers here and call the methods `snapshot` and `rewind`